### ISSUE TITLE
Flag to select unicode encoding

### ISSUE THREAD

**Issue by HFriberg** [mbr/latex#17]
build_pdf returns a a Data instance containing the generated PDF, but I see no option to set the encoding flag of the Data instance documented here:
http://pythonhosted.org/data/api.html#data.Data

In particular I get unicode errors with LANG="C" on linux which could be circumvented by explicitly setting LANG="en_US.UTF-8".. However, I could find no such solution on Windows where I still get unicode errors:
```
Traceback (most recent call last):
  File "C:\Users\eda\AppData\Roaming\Python\Python35\site-packages\data\__init__.py", line 249, in temp_saved
    self.save_to(tmp)
  File "C:\Users\eda\AppData\Roaming\Python\Python35\site-packages\data\__init__.py", line 223, in save_to
    dest.write(self.__bytes__())
  File "C:\Users\eda\AppData\Roaming\Python\Python35\site-packages\data\__init__.py", line 91, in __bytes__
    return self.readb()
  File "C:\Users\eda\AppData\Roaming\Python\Python35\site-packages\data\__init__.py", line 16, in _
    rv = f(self, *args, **kwargs)
  File "C:\Users\eda\AppData\Roaming\Python\Python35\site-packages\data\__init__.py", line 192, in readb
    return self.stream.read(*args, **kwargs)
  File "c:\local\anaconda3\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 680667: character maps to <undefined>
```

I can run pdflatex.exe on the unicode-containing tex source with no problem, so the issue is really with the Data instance.. I would like a flag in build_pdf to set the encoding explicitly..

**Comment by HFriberg-mosek** [comment]
I cloned the project and have started testing.. So far I know that simply adding encoding='utf8' didn't resolve the issue on Windows..

**Comment by mbr** [comment]
You're hitting two weak spots here: First, I don't have a Windows machine and second I've long started using UTF8 for everything to keep sane.

Here's how I believe it goes: Passed in, your Latex source gets written to disk as a text-file first. For this, Python by default uses your locale's encoding, which was C on Linux, changed to UTF8 and is CP1252 on Windows by default. It seems you have characters in your source that CP1252 (and C, i.e. ASCII) cannot represent (is it a `Ý`?)

`data` does not handle these cases of encoding confusion cleanly when writing out data. You can try the following workaround: Encode the source to UTF8 first:

```
src = "...unicode latex source here..."
pdf = build_pdf(src.encode('utf8')
```

The result will be a bytestream that does not need to be encoded in any way. Make sure your LaTeX can handle utf8.

If that is indeed the issue, it could be worth investigating whether or not `data` should apply encoding when writing out unicode data.

**Comment by mbr** [comment]
I should add: I've recently ran into issues because of the encoding of LaTeX error messages, which can be not UTF8 even if your whole system and source document are UTF8. A fix is in 362c8a07eba3, make sure you use the latest git version as this patch has not been released yet.

**Comment by HFriberg-mosek** [comment]
Well, after having set the encoding parameter everywhere in the 'latex' and 'data' packages all the way down to NamedTemporaryFile, I find that I should just have set it up front in my own code. *Duhh*

BEFORE (does not work):
```
with open(os.path.join(texpath, 'docs.tex')) as file:
    pdf = latexbld.build_pdf(file, texinputs=[texpath])
```

AFTER (works):
```
with open(os.path.join(texpath, 'docs.tex'), encoding='utf-8') as file:
    pdf = latexbld.build_pdf(file, texinputs=[texpath])
```

Thank you for your quick response!
